entproto/cmd/protoc-gen-entgrpc/converter.go

```diff
@@ -194,3 +271,30 @@ var wrapperPrimitives = map[string]string{
 	"google.protobuf.StringValue": "string",
 	"google.protobuf.BytesValue":  "[]byte",
 }
+
+func implements(r *field.RType, typ reflect.Type) bool {
```

Comment:
This is taken from ent/ent. I ported it over since I don't think it makes sense to make a public api in `ent` and might be useful for other checks here i.e. `MarshallProto` interface.